packages/excalidraw/components/SearchMenu.tsx

```diff
@@ -544,6 +544,69 @@ const normalizeWrappedText = (
   return normalizedLines.join("\n");
 };
 
+const calculateOffset = ({
+  textToStart,
+  textElement,
+  matchedWordWidth,
+  line,
+  lineNumber,
+}: {
+  textToStart: string;
+  textElement: ExcalidrawTextElement;
+  matchedWordWidth: number;
+  line: string;
+  lineNumber: number;
+}) => {
+  const startTextMeasures = measureText(
+    textToStart,
+    getFontString(textElement),
+    textElement.lineHeight,
+    true,
+  );
+
+  let offsetX = textToStart === "" ? 0 : startTextMeasures.width;
+
+  const direction = detectTextDirection(textElement.originalText);
+  const isRTL = direction === "rtl";
+
+  if (isRTL) {
+    offsetX = textElement.width - offsetX - matchedWordWidth;
```

Comment:
Haven't come with a better idea than manual language detection and calculating offset from right side